src/packages/dto/elections.dto.ts

```diff
@@ -2,6 +2,16 @@ import { ElectionStatus } from "@prisma/client";
 import { z } from "zod";
 
 export const createElectionValidator = z.object({
-  name: z.string().min(1),
+  name: z
+    .string()
+    .min(5, "O nome de usuário deve ter no mínimo 5 caracteres")
+    .max(20, "O nome de usuário deve ter no máximo 20 caracteres"),
   status: z.nativeEnum(ElectionStatus),
 });
+
+export const electionSchema = z.object({
+  name: z.string(),
+  candidates: z.array(z.string().uuid()),
+});
+
+export const updateElectionSchema = z.string().uuid();
```

Comment:
Is name in line 5 the same as name in line 13?

If so you can use:

```ts
export const electionSchema = z.object({
  name: createElectionValidator.shape.name,
  candidates: z.array(z.string().uuid()),
});
```
---
Also consider a "master" schema and then derive other schemas from it.
ex:
```ts
const masterElectionSchema = z.object({
  name: z.string(),
  id: z.string(),
  status: z.enum(),
....
})

const createElectionSchema = masterElectionSchema.omit({id: true}) // just like omit from typescript, it will have everything from masterElectionSchema excluding only the id
```

Reply:
I was thinking about reducing some repetitions in the code, thanks for this contribution.

Reply:
change here [cc51de3](https://github.com/podcodar/voting-system/pull/67/commits/cc51de33238ceb05d337b2f45d393f5644d13131)